doc/developer_guide.md

```diff
@@ -1,41 +1,189 @@
-# Developer guide
+# Developer Guide
```

Comment:
This has been copied from Panel. I should have updated it to match the param, but there could potentially be something I haven't updated correctly.